### PR TITLE
Merging creack/cleanup - proposed v2

### PR DESCRIPTION
**Pull request by jmervine**
Replacing #3 (cc @jdorfman)

@creack Merged your branch here, with the following changes...

* Adding Godeps to lock dependancies.
* Ensuring go 1.3 support, as it's latest stable for Debian (which is what I use).
* Adding LICENSE file (addressing #4)

----

Original PR notes: 

> Minor cleanup including:
> 
> * Add all example to godoc (rename example function)
> * Discard travis tests for go < 1.4
> * Use .PHONY rule in makefile
> * comply to golint
> * Use "standalone" structs instead of nested for easier testing
> * Remove pointer on header because it is already a reference
> * use http.Dump{Request,Response} instead of json.MarshalIndent
> * remove unnecessary named returns
> * remove unnecessary mutexes
> * close channels when finished
> * minor optimizations
> * Add .gitignore

### CODE REVIEW

File: maxcdn.go [hunk 1]
```diff
@@ -6,11 +6,12 @@ import (
...
 	"io/ioutil"
 	"net/http"
+	"net/http/httputil"
 	"net/url"
+	"strconv"
 	"strings"
-	"sync"
 
-	"github.com/garyburd/go-oauth/oauth"
+	"github.com/MaxCDN/go-maxcdn/Godeps/_workspace/src/github.com/garyburd/go-oauth/oauth"
```
Comment (creack) [inline maxcdn.go]: This is wrong. it should still be `"github.com/garyburd/go-oauth/oauth"`

File: maxcdn_test.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ import (
 	"net/url"
 	"testing"
 
-	. "gopkg.in/jmervine/GoT.v1"
+	. "github.com/MaxCDN/go-maxcdn/Godeps/_workspace/src/gopkg.in/jmervine/GoT.v1"
```
Comment (creack) [inline maxcdn_test.go]: This is wrong, it should be `"gopkg.in/jmervine/GoT.v1"`

**Comment by creack** [comment]
The imports should be "regular", Godeps should not be part of the import paths.
Beside this: :+1:

File: maxcdn.go (hunk 1 above)
Comment (jmervine) [inline maxcdn.go]: That's a choice in how you use Godeps, I typically use this method, opting not to change the GOPATH. That import line was written directly by Godeps itself. If we use Godeps at all, I greatly prefer this method.

**Comment by jmervine** [comment]
RE Godeps, can you provide a little insight as to why you prefer using standard import paths? I've only read the docs and not had a chance to discuss the pros and cons of both with anyone. It seemed to me that the method I used with better/simpler for both development and building, but I'd genuinely love to hear the other side of it. 

Cheers,
J

**Comment by creack** [comment]
Godep is a useful tool, but should be "hidden". The idea is to be 100% go with godep has a helper.

Forcing the import path to the godep subdir makes the import ugly but more importantly, forces the developer to use that version of the code. Which will be done anyway when using `godep go build` but having the "standard" import path allow the developer to use it's own version of the dependencies.

**Comment by jmervine** [comment]
Interesting, okay. I'll play around with that. I know there's some changes you have to make to the way travis works when attempting to use Godep without the Godep pathing.

**Comment by jdorfman** [comment]
looks good, @jmervine merge and push package whenever

**Comment by jmervine** [comment]
@jdorfman I'm merging this now. Keep in mind there's some minor breaking changes. If support gets any pings on it, point them to v1 via `import "gopkg.in/jmervine/go-maxcdn.v1"` or to to read the updated docs.

**Comment by jdorfman** [comment]
got it thanks @jmervine

**Pull request merged**